FABulous.py

```diff
@@ -0,0 +1,461 @@
+import os
+import argparse
+import re
+import sys
+import subprocess as sp
+import shutil
+import docker
+
+FABulous_root = os.getenv('FABulous_root')
+
+if FABulous_root is None:
+    print('FABulous_root is not set!')
+    print('Set FABulous_root with the following command:')
+    print('export FABulous_root=<path to FABulous root>')
+    sys.exit()
+
+
+# Create a FABulous default project that contains all the required files
+def create_project(project_dir):
+    if os.path.exists(project_dir):
+        print('Project directory already exists!')
+        sys.exit()
+    else:
+        os.mkdir(f"./{project_dir}")
+
+    os.mkdir(f"./{project_dir}/.FABulous")
+    os.mkdir(f"./{project_dir}/user_design")
+
+    shutil.copytree(f"{FABulous_root}/fabric_files/FABulous_project_template/",
+                    f"./{project_dir}/src")
+
+    shutil.copytree(
+        f"{FABulous_root}/fabric_generator/fabulous_top_wrapper_temp", f"{project_dir}/src/fabulous_top_wrapper_temp")
+
+    with open(f"{project_dir}/user_design/{project_dir.split('/')[-1]}.v", "w") as f:
+        f.write(f"module {project_dir.split('/')[-1]}(a);\n")
+        f.write(f"output a;\n")
+        f.write(f"assign a = 1;\n")
+        f.write(f"endmodule\n")
+
+
+# Generate the switch matrix of the fabric
+def switch_matrix_generation(project_dir):
+    cmd = ["python3",
```

Comment:
ideally, it'd probably be cleaner to import this as a python module and call a function (and generate nicer backtraces for debugging etc) rather than have python running python as a subpprocess